### PR TITLE
Scene Header Command 0x0B is now OK

### PR DESCRIPTION
**Pull request by Kenix3**
Knowledge gained from MM

### CODE REVIEW

File: src/code/z_scene.c [hunk 1]
```diff
@@ -279,59 +279,55 @@ void func_80098958(GlobalContext* globalCtx, SceneCmd* cmd) {
...
+        if (1) {
+            __assert("scene_info->object_bank.num <= OBJECT_EXCHANGE_BANK_MAX", "../z_scene.c", 705);
+        }
     }
 
     while (k < cmd->objectList.num) {
         nextPtr = func_800982FC(&globalCtx->objectCtx, i, *objectEntry);
         if (i < OBJECT_EXCHANGE_BANK_MAX - 1) {
-            globalCtx->objectCtx.status[i + 1].segment = nextPtr;
+            (globalCtx->objectCtx.status + i + 1)->segment = nextPtr;
```
Comment (Roman971) [inline src/code/z_scene.c:322]: Is that necessary? Imo the array access is much cleaner.

File: src/code/z_scene.c [hunk 2]
```diff
@@ -279,59 +279,55 @@ void func_80098958(GlobalContext* globalCtx, SceneCmd* cmd) {
...
+        
+        i++;
+        k++;
+        objectEntry++;
+        status++;
     }
 
     if (cmd->objectList.num > OBJECT_EXCHANGE_BANK_MAX) {
-        __assert("scene_info->object_bank.num <= OBJECT_EXCHANGE_BANK_MAX", "../z_scene.c", 705);
+        if (1) {
```
Comment (Roman971) [inline src/code/z_scene.c:314]: Can this if (1) be empty? Possibly after moving it before or after this block.

File: src/code/z_scene.c (hunk 1 above)
Comment (Kenix3) [reply src/code/z_scene.c:322]: This breaks regalloc.

File: src/code/z_scene.c (hunk 2 above)
Comment (Kenix3) [reply src/code/z_scene.c:314]: This can match with empty if (1)

File: src/code/z_scene.c [hunk 3]
```diff
@@ -279,59 +279,55 @@ void func_80098958(GlobalContext* globalCtx, SceneCmd* cmd) {
...
         }
+        
+        i++;
+        k++;
+        objectEntry++;
+        status++;
     }
 
     if (cmd->objectList.num > OBJECT_EXCHANGE_BANK_MAX) {
+        if (1) {}
```
Comment (fig02) [inline src/code/z_scene.c:314]: i believe the only thing holding this up is that we wanted to try putting the if (1) {} outside of the assert condition. does that still match?

File: src/code/z_scene.c [hunk 4]
```diff
@@ -279,59 +279,58 @@ void func_80098958(GlobalContext* globalCtx, SceneCmd* cmd) {
...
         __assert("scene_info->object_bank.num <= OBJECT_EXCHANGE_BANK_MAX", "../z_scene.c", 705);
     }
+    
+    if (1) {}
 
     while (k < cmd->objectList.num) {
         nextPtr = func_800982FC(&globalCtx->objectCtx, i, *objectEntry);
         if (i < OBJECT_EXCHANGE_BANK_MAX - 1) {
-            globalCtx->objectCtx.status[i + 1].segment = nextPtr;
+            // This pointer math matches while array access does not
```
Comment (fig02) [inline src/code/z_scene.c:324]: looks good, just need to remove the comment

**Pull request merged**